### PR TITLE
[Doc][CARBONDATA-4185] Doc Changes for Heterogeneous format segments in carbondata

### PR DESCRIPTION
**Pull request by maheshrajus**
### Why is this PR needed?
 Heterogeneous format segments in carbondata documenation.

 ### What changes were proposed in this PR?
Add segment feature background and impact on existed carbondata features

 ### Does this PR introduce any user interface change?
 - No
 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12602/job/ApacheCarbon_PR_Builder_2.4.5/3609/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5354/

File: docs/addsegment-guide.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+-->
+
+# Heterogeneous format segments in carbondata
+
+###Background
+In the industry, many users already adopted to data with different formats like ORC, Parquet, JSON, CSV etc.,  
+If users want to migrate to Carbondata for better performance or for better features then there is no direct way. 
+All the existing data needs to be converted to Carbondata to migrate.  
+This solution works out if the existing data is less, what if the existing data is more?   
+This process of converting is very time consuming and error-prone.
```
Comment (kunal642) [inline docs/addsegment-guide.md:25]: Line 25 can be changed to "Heterogeneous format segments aims to solve this problem by avoiding data conversion"

File: docs/addsegment-guide.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+###Add segment with path and format
+To solve the above problem carbon introduces the heterogeneous format segments which can read the data as per the mentioned format of the segment.
```
Comment (kunal642) [inline docs/addsegment-guide.md:28]: after line 25 is changed "To solve the above problem carbon introduces the" can be removed in line 28

File: docs/addsegment-guide.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+###Changes to Presto Integration
+CarbondataSplitManager can create the splits for carbon and as well as for other formats and 
+ choose the page source as per the split.  
+
+### Impact on existed feature
+**Count(\*) query:**  In case if the segments are mixed with different formats then driver side
+ optimization for count(*) query will not work so it will be executed on executor side.
+
+**INdex DataMaps:** Datamaps like block/blocklet datamap will only work for carbondata format
```
Comment (kunal642) [inline docs/addsegment-guide.md:62]: change "INdex" to "Index"

File: docs/addsegment-guide.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+**Compaction:** The other format segments cannot be compacted but carbondata segments inside that
+ table will be compacted.
+
+**Show Segments:** Now it shows the format and path of the segment along with current information.
+
+**Delete Segments & Clean Files:**  If the segment to be deleted is external then it will not be
+ deleted physically. If the segment is present internally only will be deleted physically.
+
+**MV/Preaggregate DataMap:** These datamaps can be created on the mixed format table without any
```
Comment (kunal642) [inline docs/addsegment-guide.md:76]: No need to mention "Preaggregate" now

File: docs/addsegment-guide.md (hunk 1 above)
Comment (maheshrajus) [reply docs/addsegment-guide.md:25]: done

File: docs/addsegment-guide.md (hunk 2 above)
Comment (maheshrajus) [reply docs/addsegment-guide.md:28]: done

File: docs/addsegment-guide.md (hunk 3 above)
Comment (maheshrajus) [reply docs/addsegment-guide.md:62]: done

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12602/job/ApacheCarbonPRBuilder2.3/5407/

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**